numpy/core/src/multiarray/lowlevel_strided_loops.c.src

```diff
@@ -206,6 +206,9 @@ static NPY_GCC_OPT_3 void
 #else
     npy_uint64 temp0, temp1;
 #endif
+    if (N == 0) {
```

Comment:
Could probably do with a comment along the lines of "dereferencing `src` when `N == 0` is not allowed